operators/src/source/ogr_source/dataset_iterator.rs

```diff
@@ -119,16 +119,69 @@ impl OgrDatasetIterator {
         query_rectangle: &VectorQueryRectangle,
         attribute_filters: &[AttributeFilter],
     ) -> Result<FeaturesProvider<'d>> {
-        // TODO: add OGR time filter if forced
+        let filter_string = if dataset.driver().short_name() == "CSV" {
+            FeaturesProvider::create_attribute_filter_string_cast(attribute_filters)
+        } else {
+            FeaturesProvider::create_attribute_filter_string(attribute_filters)
+        };
+
+        let time_filter = if dataset_information.force_ogr_time_filter {
+            debug!(
+                "using time filter {:?} for layer {:?}",
+                query_rectangle.time_interval, &dataset_information.layer_name
+            );
+            FeaturesProvider::create_time_filter_string(
+                dataset_information.time.clone(),
+                query_rectangle.time_interval,
+                &dataset.driver().short_name(),
+            )
+        } else {
+            None
+        };
+
+        let final_filter = filter_string
+            .map(|f| match &dataset_information.attribute_query {
+                Some(a) => format!("({a}) AND {f}"),
+                None => f,
+            })
+            .or_else(|| dataset_information.attribute_query.clone())
+            .map(|f| match &time_filter {
+                None => f,
+                Some(t) => format!("({t}) AND {f}"),
+            })
+            .or_else(|| time_filter.clone());
 
         let mut features_provider = if let Some(sql) = dataset_information.sql_query.as_ref() {
+            let query = if let Some(filter) = final_filter {
+                debug!(
+                    "using attribute filter {:?} for layer {:?}",
+                    &filter, &dataset_information.layer_name
+                );
+
+                // This is necessary because otherwise the GDAL postgres driver does not perform a filter-pushdown in case an explicit SQL query is given
```

Comment:
what about the spatial filter? does that not interfere with the sql_query?

Reply:
As mentioned in another comment above, for the spatial filter GDAL uses the SQL query as an inner query and adds a spatial filter to the outer query. It does not do this for other filter conditions (attribute filters, where I include the time filter), so this is done manually